frontend/src/component/common/BreadcrumbNav/BreadcrumbNav.tsx

```diff
@@ -50,55 +50,57 @@ const BreadcrumbNav = () => {
         );
 
     return (
-        <ConditionallyRender
-            condition={
-                (location.pathname.includes('admin') && isAdmin) ||
-                !location.pathname.includes('admin')
-            }
-            show={
-                <ConditionallyRender
-                    condition={paths.length > 1}
-                    show={
-                        <StyledBreadCrumbs aria-label="Breadcrumbs">
-                            {paths.map((path, index) => {
-                                const lastItem = index === paths.length - 1;
-                                if (lastItem) {
+        <StyledBreadcrumbContainer>
+            <ConditionallyRender
+                condition={
+                    (location.pathname.includes('admin') && isAdmin) ||
+                    !location.pathname.includes('admin')
+                }
+                show={
+                    <ConditionallyRender
+                        condition={paths.length > 1}
+                        show={
+                            <Breadcrumbs aria-label="Breadcrumbs">
+                                {paths.map((path, index) => {
+                                    const lastItem = index === paths.length - 1;
+                                    if (lastItem) {
+                                        return (
+                                            <StyledParagraph key={path}>
+                                                <StringTruncator
+                                                    text={path}
+                                                    maxWidth="200"
+                                                    maxLength={25}
+                                                />
+                                            </StyledParagraph>
+                                        );
+                                    }
+
+                                    let link = '/';
+
+                                    paths.forEach((path, i) => {
+                                        if (i !== index && i < index) {
+                                            link += path + '/';
+                                        } else if (i === index) {
+                                            link += path;
+                                        }
+                                    });
```

Comment:
Just for fun, this is a for-loop (`paths.map`) inside a for-loop (`paths.forEach`), so it's `O(length(path)²)`. Perhaps, this can be done with a an accumulator (for the link) and make it linear.

But not a blocker for the PR as this is the same the previous code was doing

Reply:
Also, I assume lenght is usually no longer than 5

Reply:
Outside of the scope of this PR, but I agree that we could improve this 👍